### PR TITLE
doc: adding treemacs readme

### PR DESCRIPTION
**Pull request by jeetelongname**
This is a stub I need to finish but I wanted to upload it for... I don't really know. Also its pertinent to #1166

### CODE REVIEW

**Comment by jeetelongname** [comment]
I am handing in my work for marking

File: modules/ui/treemacs/README.org [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  - [[#maintainers][Maintainers]]
+  - [[#module-flags][Module Flags]]
+  - [[#plugins][Plugins]]
+- [[#prerequisites][Prerequisites]]
+- [[#features][Features]]
+- [[#configuration][Configuration]]
+- [[#troubleshooting][Troubleshooting]]
+
+* Description
+This module adds the treemacs project sidebar
```
Comment (polyedre) [inline modules/ui/treemacs/README.org:17]: I would have added some punctuation here.

```diff
- This module adds the treemacs project sidebar
+ This module adds the treemacs project sidebar.
```

File: modules/ui/treemacs/README.org [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
++ =:editor evil +everywhere=
+  + [[https://github.com/Alexander-Miller/treemacs/#treemacs-evil][treemacs-evil]]
++ =:tools magit=
+  + [[https://github.com/Alexander-Miller/treemacs/#treemacs-magit][treemacs-magit]]
++ =:ui workspaces=
+  + [[https://github.com/Alexander-Miller/treemacs/#treemacs-persptreemacs-perspective][treemacs-persp]]
+
+* Prerequisites
+If ~+treemacs-git-mode~ is set to ~extended~ or ~deferred~ you will need to have
+~python3~ installed
```
Comment (polyedre) [inline modules/ui/treemacs/README.org:46]: I would have added some punctuation here.

```diff
- ~python3~ installed
+ ~python3~ installed.
```

File: modules/ui/treemacs/README.org [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+
+* Features
+The project bar can be opened using =SPC o p=
```
Comment (polyedre) [inline modules/ui/treemacs/README.org:49]: Same here!

```diff
- The project bar can be opened using =SPC o p=
+ The project bar can be opened using =SPC o p=.
```

File: modules/ui/treemacs/README.org [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+| Keybind | Description                                                     |
+|---------+-----------------------------------------------------------------|
+| =o s=     | Open a horizontal dired buffer on the currently highlighted node |
+| =o v=     | Open a vertical dired buffer on the currently highlighted node  |
+
+* Configuration
+the variable ~+treemacs-git-mode~ sets the type of git integration for treemacs
```
Comment (polyedre) [inline modules/ui/treemacs/README.org:57]:
```diff
- the variable ~+treemacs-git-mode~ sets the type of git integration for treemacs
+ The variable ~+treemacs-git-mode~ sets the type of git integration for treemacs.
```

Comment (polyedre) [reply modules/ui/treemacs/README.org:57]: This sentence was not really clear at first but the 3 bullets points helps a lot!

**Review by polyedre — COMMENTED** [review]
This looks good to me!

**Comment by polyedre** [comment]
Thanks for the fixes! It seems that the file https://github.com/hlissner/doom-emacs/blob/develop/modules/ui/treemacs/README.org is already existing. You have additions however so it would be great if you could rebase your PR!

**Comment by jeetelongname** [comment]
I am aware. Thanks for the reminder! Ill probably rebase it later today /tomorrow

**Comment by jeetelongname** [comment]
Ok that was a bit of a mess but it should be rebased properly now

**Review by polyedre — APPROVED** [review]

**Review by hlissner — APPROVED** [review]

**Comment by hlissner** [comment]
Thanks!

**Pull request merged**